imports/api/parcels/publications.js

```diff
@@ -25,6 +25,9 @@ Meteor.publish('parcels.ofSelf', function parcelsOfSelf(params) {
   const { communityId } = params;
   if (!Meteor.user()) return this.ready();
   const parcelIds = Meteor.user().memberships(communityId).map(m => m.parcelId);
+  parcelIds.forEach((parcelId) => {
+    Parcels.findOne(parcelId).forEachLed(parcel => parcelIds.push(parcel._id));
```

Comment:
Ez igy egy veszelyes jatek, altalaban komoly galibat szokott okozni, ha egy tomb ertekeibe akozben nyulkalok bele, mikozben epp iteralok rajta. Gondold vegig, a kulso forEach lepked a tomb elemein, es minden lepesnel belul uj elemeket pakolunk abba a tombbe, amin a forEach probal vegigmenni.
Jelen esetben valosyinuleg csak annyi tortenik, hogy a ledParcelek menet kozben hozzatoldodnak, es azokon is vegig fogunk vegen iteralni. De biztonsagosabb ha ilyet nem csinalunk. (nem irjuk azt amit iteralunk)

Reply:
Upsz, de jó hogy figyelsz, erre nem is gondoltam.

Reply:
Nem kerul ez amugy most vegtelen ciklusba?
Mert a forEachLed -ben benne van a parcel sajat maga, tehat ahogy iteralok, es parcelId-re kerul a sor, hozza fogom push-olni parcleId-t a tomb vegehez, tehat ujra sorra fog kerulni, amikor is megint hozza fogom push-olni a tomb vegehez, stb. Arra tippelnek hogy vegtelen ciklus.
Ha nem lenne a "Led" halmazba sajat maga is benne, akkor viszont meg akar jol is mukodhet, ha mindig tenylegesen uj elemeket push-olna a tomb vegere.

Reply:
Egyebkent a javitas csak annyi, hogy nem ugyanabba, hanem egy uj "ledParcelIds" tombbe kene pusholni az eredmenyt, es azt lehet is publikalni.

Reply:
Nem kerül végtelen ciklusba, mert: The range of elements processed by forEach() is set before the first invocation of callback. Elements which are appended to the array after the call to forEach() begins will not be visited by callback. If existing elements of the array are changed, or deleted, their value as passed to callback will be the value at the time forEach() visits them; elements that are deleted before being visited are not visited. If elements that are already visited are removed (e.g. using shift()) during the iteration, later elements will be skipped.

Reply:
Azta defuran viselkedik a forEach a javascriptben.
Akkor viszont jo igy ahogy csinaltad.
De szerencse hogy van a csapatban aki tudja hogyan mukodik a javascript. :)
nekem meg mindig okoz meglepetseket